lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -115,17 +122,33 @@ void pthread_scheduler_wait_cq (cl_command_queue cq)
           return;
         }
       POCL_UNLOCK_OBJ (cq);
-      pthread_cond_wait (&scheduler.cq_finished_cond,
-                         &scheduler.cq_finished_lock);
-      pthread_mutex_unlock (&scheduler.cq_finished_lock);
+
+#ifdef HAVE_CLOCK_GETTIME
```

Comment:
it's a workaround, the pthread driver sometimes gets stuck in the loop waiting on cond_wait for queue to empty while it's actually empty. This patch just makes the wait time-limited. I'll add a comment.